app/src/pages/inside/settingsPage/settingTabs/demoDataTab/demoDataTabForm/demoDataTabForm.jsx

```diff
@@ -0,0 +1,135 @@
+    id: 'DemoDataTabForm.preloaderInfo',
+    defaultMessage:
+      'Data generation is started. The process can take several minutes, please wait.',
+  },
+  generateDemoDataSuccess: {
+    id: 'SuccessMessages.generateDemoDataSuccess',
+    defaultMessage: 'Demo data has been generated',
+  },
+});
+
+@reduxForm({
+  form: 'demoDataTabForm',
+  validate: ({ demoDataPostfix }) => ({
+    demoDataPostfix:
+      (!demoDataPostfix || !validate.demoDataPostfix(demoDataPostfix)) && 'demoDataPostfixHint',
+  }),
+})
+@connect(
+  (state) => ({
+    projectId: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+@injectIntl
+export class DemoDataTabForm extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    projectId: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    reset: PropTypes.func.isRequired,
+  };
+
+  state = {
+    isLoading: false,
+  };
+
+  onFormSubmit = (formData) => {
+    const { intl, projectId, reset } = this.props;
+    const data = {
+      isCreateDashboard: 'true',
+      postfix: formData.demoDataPostfix,
+    };
+
+    this.setState({
+      isLoading: true,
+    });
+    fetch(URLS.generateDemoData(projectId), { method: 'POST', data })
+      .then(() => {
+        this.props.showNotification({
+          message: intl.formatMessage(messages.generateDemoDataSuccess),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.setState({
+          isLoading: false,
+        });
+        reset('demoDataTabForm');
+      })
+      .catch((e) => {
+        const error = (e.response && e.response.data && e.response.data.message) || e.message;
+        this.props.showNotification({
+          messageId: 'failureDefault',
+          type: NOTIFICATION_TYPES.ERROR,
+          values: { error },
+        });
+        this.setState({
+          isLoading: false,
+        });
+      });
+  };
+
+  render() {
+    const { intl, handleSubmit } = this.props;
+
+    return (
+      <form onSubmit={handleSubmit(this.onFormSubmit)}>
+        <div className={cx('input-wrapper')}>
+          <FieldProvider name="demoDataPostfix" disabled={this.state.isLoading}>
+            <FieldErrorHint>
+              <Input placeholder={intl.formatMessage(messages.postfixInputPlaceholder)} />
+            </FieldErrorHint>
+          </FieldProvider>
+        </div>
+        <div className={cx('generate-button-container')}>
+          <div className={cx('generate-button')}>
+            <BigButton mobileDisabled type="submit" disabled={this.state.isLoading}>
+              <span className={cx('generate-button-title')}>
+                {intl.formatMessage(messages.generateButtonTitle)}
+              </span>
+            </BigButton>
+          </div>
+        </div>
+        {this.state.isLoading && (
+          <div className={cx('preloader-wrapper')}>
+            <div className={cx('preloader-icon')}>
```

Comment:
You don't need in 'preloader-icon' div

Reply:
This is necessary for the correct order of the elements